src/cluster.py

```diff
@@ -95,6 +122,62 @@ def _get_postgresql_version(self) -> str:
         # Remove the Ubuntu revision from the version.
         return str(package.version).split("+")[0]
 
+    def get_primary(self, unit_name_pattern=False) -> str:
+        """Get primary instance.
+
+        Args:
+            unit_name_pattern: whether to convert pod name to unit name
+
+        Returns:
+            primary pod or unit name.
+        """
+        primary = None
+        # Request info from cluster endpoint (which returns all members of the cluster).
+        r = requests.get(f"http://{self.unit_ip}:8008/cluster")
```

Comment:
```suggestion
        cluster_status = requests.get(f"http://{self.unit_ip}:8008/cluster")
```

Reply:
Variable renamed in https://github.com/canonical/postgresql-operator/pull/10/commits/0afd3c85851e1dc5c9e3350dd29cbc130b28c056.